### PR TITLE
PIM-4964: Use enable / disable import parameter only to create the product

### PR DESCRIPTION
**Pull request by solivier**
| Q                 | A
| ----------------- | ---
| Specs             | Yes
| Behats            | Yes
| Blue CI           | Yes
| Changelog updated | Yes
| Review and 2 GTM  | No
| Micro Demo (PO)   | Yes

### CODE REVIEW

File: spec/Pim/Component/Connector/Processor/Denormalization/ProductProcessorSpec.php [hunk 1]
```diff
@@ -1035,4 +1035,108 @@ function it_skips_a_product_when_there_is_nothing_to_update_with_localized_value
...
+
+        $originalData = [
+            'sku' => 'tshirt',
+            'family' => 'TShirt',
+            'description-en_US-mobile' => 'My description',
+            'name-fr_FR' => 'T-shirt super beau',
+            'name-en_US' => 'My awesome T-shirt',
+            'enabled' => 0
+        ];
+        $convertedData =                 [
```
Comment (skeleton) [inline spec/Pim/Component/Connector/Processor/Denormalization/ProductProcessorSpec.php]: extra spaces

File: spec/Pim/Component/Connector/Processor/Denormalization/ProductProcessorSpec.php [hunk 2]
```diff
@@ -1035,4 +1035,108 @@ function it_skips_a_product_when_there_is_nothing_to_update_with_localized_value
...
+            ->shouldBeCalled();
+
+        $productValidator
+            ->validate($product)
+            ->willReturn($violationList);
+
+        $this
+            ->process($originalData)
+            ->shouldReturn($product);
+
```
Comment (skeleton) [inline spec/Pim/Component/Connector/Processor/Denormalization/ProductProcessorSpec.php]: extra blank line

**Comment by skeleton** [comment]
2 baby comments, then GTM

File: features/import/product/import_products.feature [hunk 3]
```diff
@@ -195,3 +195,28 @@ Feature: Execute a job
...
+      SKU-003;Renault;French touch for tractors;1
+      SKU-004;New Holland;Faster tractors;1
+      """
+    And the following job "footwear_product_import" configuration:
+      | filePath | %file to import% |
+    When I am on the "footwear_product_import" import job page
+    And I launch the import job
+    And I wait for the "footwear_product_import" job to finish
+    Then there should be 4 products
+    And product "SKU-001" should be disabled
```
Comment (ClementGautier) [inline features/import/product/import_products.feature]: I'm not sure that it is what the PO asked. In your case here the file to import should set the enabled flag to true.

Comment (solivier) [inline features/import/product/import_products.feature]: fixed

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php [hunk 4]
```diff
@@ -48,6 +48,9 @@ class ProductProcessor extends AbstractProcessor
     /** @var bool */
     protected $enabled = true;
 
+    /** @var bool */
+    protected $hasStatus = false;
```
Comment (jjanvier) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: what means hasStatus?

Comment (solivier) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: it's when the imported file has the product status

File: features/import/product/import_products.feature [hunk 5]
```diff
@@ -195,3 +195,55 @@ Feature: Execute a job
...
+      | filePath | %file to import% |
+      | enabled  | no               |
+    When I am on the "footwear_product_import" import job page
+    And I launch the import job
+    And I wait for the "footwear_product_import" job to finish
+    Then there should be 4 products
+    And product "SKU-001" should be disabled
+    And product "SKU-002" should be enabled
+    And product "SKU-003" should be disabled
+    And product "SKU-004" should be disabled
```
Comment (solivier) [inline features/import/product/import_products.feature]: missing line

File: src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php [hunk 6]
```diff
@@ -381,6 +388,10 @@ protected function filterIdenticalData(ProductInterface $product, array $filtere
      */
     protected function convertItemData(array $item)
     {
+        if (array_key_exists('enabled', $item)) {
```
Comment (phaseinducer) [inline src/Pim/Component/Connector/Processor/Denormalization/ProductProcessor.php]: `$this->itemHasStatus = array_key_exists('enabled', $item);` ?

**Comment by phaseinducer** [comment]
GTM

**Pull request merged**